internal/controller/grafanaorganization_controller.go

```diff
@@ -99,17 +101,79 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 			return ctrl.Result{}, errors.WithStack(err)
 		}
 	}
-	_, err := grafanaAPI.Orgs.UpdateOrg(1, &grafanaAPIModels.UpdateOrgForm{
-		Name: "Shared Org.",
-	})
+
+	// if the CR doesn't have an orgID, create the organization in Grafana and update the status
+	if grafanaOrganization.Status.OrgID == "" {
+		return ctrl.Result{}, r.createOrganizationInGrafana(ctx, grafanaAPI, grafanaOrganization)
+	} else {
+		// Cast orgID to int64
+		orgID, err := strconv.ParseInt(grafanaOrganization.Status.OrgID, 10, 64)
+		if err != nil {
+			logger.Error(err, "Failed to parse OrgID")
+			return ctrl.Result{}, errors.WithStack(err)
+		}
+
+		grievious, err := grafanaAPI.Orgs.GetOrgByID(orgID)
+		if err != nil {
+			logger.Error(err, "Failed to get organization by ID")
```

Comment:
Why not create it here instead?

Reply:
Maybe the status is old and we should replace it?

Reply:
Hmm maybe but we still need to check whether the name is available

Reply:
Yes that's why I think having the name check first is better :)